OpenRA.Game/Scripting/ScriptObjectWrapper.cs

```diff
@@ -30,10 +30,11 @@ public ScriptObjectWrapper(ScriptContext context)
 
 		protected void Bind(IEnumerable<object> clrObjects)
 		{
+			bool isAIScriptBinding = Context is AIScriptContext;
```

Comment:
I'll work on this issue this weekend. What made me do the check is that in one class, some functions are AISafe and others are not. What is in my mind now is that to make a custom parser that generates ai script binding C# classes from current map script binding classes, which will ignore the functions inside "#aiunsafe" regions.